src/library/filescanner_ffmpeg.c

```diff
@@ -111,7 +112,7 @@ parse_date(struct media_file_info *mfi, char *date_string)
        || strptime(date_string, "%F", &tm)
      )
     {
-      *date_released = (uint32_t)mktime(&tm);
+      *date_released = (int32_t)mktime(&tm);
```

Comment:
Not quite sure why it's int32_t here? (and below)

Reply:
Do you mean why not let the compiler force the implicit conversion to the signed int64 ?

It has to be signed to deal with negative values meaning dates prior 1970.  I kept to 32bit as somehow the existing code has some assumptions that mktime needs to 32bit. Maybe an oversight on my part and we don't need the explo cast

Reply:
drop cast since its redundant

Reply:
C's integer rules always confuse me, but I'm thinking that since mktime returns time_t and you want to store it as int64 a cast to int64 would be needed?

Reply:
This has always been my understanding, aklthough without the correct terminology i'm can't find the right reference online

_promotion_ rules will take care of assignment of smaller (say 32bit `time_t`) RHS to larger (64bit) LHS, no cast needed since LHS is _wider_.  If both LHS / RHS are same size but different sign-ness then result is on bitwise representation of LHS destination.

 i can add explicit cast to `int64_t` in the assigment but the assignment to `int64` should be enough